xiaomi.vacuum.gen1/firmwarebuilder/imagebuilder.sh

```diff
@@ -18,146 +18,200 @@
 # place english.pkg and v11_<fw_version>.pkg (e.g. v11_003077.pkg) in this folder
 # place your authorized_keys in this folder
 #
-IS_MAC=false
-FIRMWARE_VERSION=$1
-SOUNDFILE=english.pkg
 
-if [[ ! -n "$FIRMWARE_VERSION" ]]; then
-	echo "You need to specify a firmware version, e.g. 003094"
+if [[ $EUID -ne 0 ]]; then
+	echo "You must be a root user" 2>&1
 	exit 1
 fi
 
+IS_MAC=false
 if [[ $OSTYPE == darwin* ]]; then
 	# Mac OSX
 	IS_MAC=true
 	echo "Running on a Mac, adjusting commands accordingly"
 fi
 
-if [ ! -f /usr/bin/ccrypt ] && ! $IS_MAC; then
-    echo "Ccrypt not found! Please install it (e.g. by apt-get install ccrypt)"
+if [ ! -f /usr/bin/ccrypt -a "$IS_MAC" = false ]; then
+    echo "Ccrypt not found! Please install it (e.g. by apt install ccrypt)"
 	exit 1
 fi
 
-if [ ! -f /usr/local/bin/ccrypt ] && $IS_MAC; then
+if [ ! -f /usr/local/bin/ccrypt -a "$IS_MAC" = true ]; then
     echo "Ccrypt not found! Please install it (e.g. by brew install ccrypt)"
 	exit 1
 fi
 
-if [ ! -f $SOUNDFILE ]; then
-    echo "File $SOUNDFILE not found!"
+if [[ $# -eq 0 ]]; then
+	cat << EOF
+usage: sudo ./firmwarebuilder -f v11_003094.pkg [-s english.pkg] [-k authorized_keys] [ -t Europe/Berlin ] [--disable-xiaomi]
```

Comment:
I like this online doku! For Ruby there's a gem called [thor](https://github.com/erikhuda/thor) which creates this documentation on the fly. Maybe there's something similar for bash?

Reply:
nah, bash is straightforward. you do manually what you want :)

Reply:
just a suggestion ;) I wanted to add something similar when I added the parameter for the version but my bash knowledge is limited so I would have used something like the mentioned gem